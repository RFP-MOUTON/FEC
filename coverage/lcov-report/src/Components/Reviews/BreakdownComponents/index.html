
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/Components/Reviews/BreakdownComponents</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../index.html">All files</a> src/Components/Reviews/BreakdownComponents</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">9.52% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>4/42</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/20</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/7</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">9.76% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>4/41</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Rating.jsx"><a href="Rating.jsx.html">Rating.jsx</a></td>
	<td data-value="4.35" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 4%"></div><div class="cover-empty" style="width: 96%"></div></div>
	</td>
	<td data-value="4.35" class="pct low">4.35%</td>
	<td data-value="23" class="abs low">1/23</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="4.55" class="pct low">4.55%</td>
	<td data-value="22" class="abs low">1/22</td>
	</tr>

<tr>
	<td class="file low" data-value="RecommendedComponent.jsx"><a href="RecommendedComponent.jsx.html">RecommendedComponent.jsx</a></td>
	<td data-value="20" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 20%"></div><div class="cover-empty" style="width: 80%"></div></div>
	</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	</tr>

<tr>
	<td class="file low" data-value="SortingListComponent.jsx"><a href="SortingListComponent.jsx.html">SortingListComponent.jsx</a></td>
	<td data-value="11.11" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 11%"></div><div class="cover-empty" style="width: 89%"></div></div>
	</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="9" class="abs low">1/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="9" class="abs low">1/9</td>
	</tr>

<tr>
	<td class="file low" data-value="StarbarComponent.jsx"><a href="StarbarComponent.jsx.html">StarbarComponent.jsx</a></td>
	<td data-value="20" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 20%"></div><div class="cover-empty" style="width: 80%"></div></div>
	</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Tue Mar 09 2021 18:43:43 GMT-0800 (Pacific Standard Time)
            </div>
        </div>
        <script src="../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../sorter.js"></script>
        <script src="../../../../block-navigation.js"></script>
    </body>
</html>
    