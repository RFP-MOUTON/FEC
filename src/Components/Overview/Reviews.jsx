import React from 'react';
import axios from 'axios';
import instance from './instance';

class Reviews extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return <div>Greetings</div>;
  }
}

export default Reviews;
