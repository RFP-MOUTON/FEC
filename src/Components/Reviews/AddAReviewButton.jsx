import React from 'react';

import AddAReviewForm from './AddAReviewModal.jsx';

class AddAReview extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modalVisible: false,
    };
    this.toggleModal = this.toggleModal.bind(this);
  }

  toggleModal() {
    this.setState((previousState) => {
      return {
        modalVisible: !previousState.modalVisible,
      };
    });
  }

  render() {
    const { modalVisible } = this.state;
    return (
      <>
        <button id="AddAReviewButton" type="button" onClick={this.toggleModal}>
          ADD A REVIEW
        </button>
        {modalVisible ? <AddAReviewForm toggle={this.toggleModal} /> : null}
      </>
    );
  }
}

export default AddAReview;
