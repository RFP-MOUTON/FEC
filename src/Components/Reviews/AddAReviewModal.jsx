/* eslint-disable react/jsx-boolean-value */
/* eslint-disable object-shorthand */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import axios from 'axios';

import SizeComponent from './CharactersisticsFormComponents/SizeSelector.jsx';
import WidthComponent from './CharactersisticsFormComponents/WidthSelector.jsx';
import ComfortComponent from './CharactersisticsFormComponents/ComfortSelector.jsx';
import QualityComponent from './CharactersisticsFormComponents/QualitySelector.jsx';
import LengthComponent from './CharactersisticsFormComponents/LengthSelector.jsx';
import FitComponent from './CharactersisticsFormComponents/FitSelector.jsx';

class AddAReviewForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      rating: 1,
      ratingMessage: 'Poor',
      recommend: null,
      size: null,
      width: null,
      comfort: null,
      quality: null,
      length: null,
      fit: null,
      summary: '',
      body: '',
      photo: '',
      name: '',
      email: '',
    };
    this.handleStarClick = this.handleStarClick.bind(this);
    this.handleRecommend = this.handleRecommend.bind(this);
    this.handleSize = this.handleSize.bind(this);
    this.handleComfort = this.handleComfort.bind(this);
    this.handleQuality = this.handleQuality.bind(this);
    this.handleLength = this.handleLength.bind(this);
    this.handleWidth = this.handleWidth.bind(this);
    this.handleFit = this.handleFit.bind(this);
    this.handleSummary = this.handleSummary.bind(this);
    this.handleBody = this.handleBody.bind(this);
    this.handlePhoto = this.handlePhoto.bind(this);
    this.handleName = this.handleName.bind(this);
    this.handleEmail = this.handleEmail.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleStarClick(event) {
    let ratingMessage = '';
    if (Number(event.target.dataset.rating) === 1) {
      ratingMessage = 'Poor';
    }
    if (Number(event.target.dataset.rating) === 2) {
      ratingMessage = 'Fair';
    }
    if (Number(event.target.dataset.rating) === 3) {
      ratingMessage = 'Average';
    }
    if (Number(event.target.dataset.rating) === 4) {
      ratingMessage = 'Good';
    }
    if (Number(event.target.dataset.rating) === 5) {
      ratingMessage = 'Great';
    }
    this.setState({
      rating: event.target.dataset.rating,
      ratingMessage,
    });
  }

  handleRecommend(event) {
    this.setState({
      recommend: event.target.value,
    });
  }

  handleSize(event) {
    this.setState({
      size: event.target.value,
    });
  }

  handleWidth(event) {
    this.setState({
      width: event.target.value,
    });
  }

  handleComfort(event) {
    this.setState({
      comfort: event.target.value,
    });
  }

  handleQuality(event) {
    this.setState({
      quality: event.target.value,
    });
  }

  handleLength(event) {
    this.setState({
      length: event.target.value,
    });
  }

  handleFit(event) {
    this.setState({
      fit: event.target.value,
    });
  }

  handleSummary(event) {
    this.setState({
      summary: event.target.value,
    });
  }

  handleBody(event) {
    this.setState({
      body: event.target.value,
    });
  }

  handlePhoto(event) {
    this.setState({
      photo: event.target.value,
    });
  }

  handleName(event) {
    this.setState({
      name: event.target.value,
    });
  }

  handleEmail(event) {
    this.setState({
      email: event.target.value,
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const { metaData, submit } = this.props;
    // eslint-disable-next-line camelcase
    const { product_id, characteristics } = metaData;
    const { rating, summary, body, recommend, name, email, photo } = this.state;
    const newCharacteristics = {};
    let newRec = true;
    if (recommend === 'true') {
      newRec = true;
    } else {
      newRec = false;
    }

    Object.keys(characteristics).forEach((key) => {
      const current = characteristics[key];
      // eslint-disable-next-line react/destructuring-assignment
      newCharacteristics[current.id] = Number(this.state[key.toLowerCase()]);
    });

    const reviewObject = {
      product_id: Number(product_id),
      rating: Number(rating),
      summary: summary,
      body: body,
      recommend: newRec,
      name: name,
      email: email,
      photos: [photo],
      characteristics: newCharacteristics,
    };

    axios.post('/reviews', reviewObject).then(() => {
      submit();
    });
  }

  render() {
    const { toggle, metaData, productName } = this.props;
    const {
      rating,
      ratingMessage,
      summary,
      body,
      photo,
      name,
      email,
    } = this.state;
    const { characteristics } = metaData;
    const { Width, Size, Comfort, Quality, Fit, Length } = characteristics;
    return (
      <div className="blur">
        <form id="addAReviewModal">
          <div id="formLine1">Write Your Review</div>
          <div id="formLine2">For the {productName}</div>
          <div id="overallRating">
            <div className="overallMessage">Overall rating*</div>

            <div
              className="formStar"
              data-rating="1"
              onClick={this.handleStarClick}
              style={{ background: rating > 0 ? 'black' : null }}
            />
            <div
              className="formStar"
              data-rating="2"
              onClick={this.handleStarClick}
              style={{ background: rating > 1 ? 'black' : null }}
            />
            <div
              className="formStar"
              data-rating="3"
              onClick={this.handleStarClick}
              style={{ background: rating > 2 ? 'black' : null }}
            />
            <div
              className="formStar"
              data-rating="4"
              onClick={this.handleStarClick}
              style={{ background: rating > 3 ? 'black' : null }}
            />
            <div
              className="formStar"
              data-rating="5"
              onClick={this.handleStarClick}
              style={{ background: rating > 4 ? 'black' : null }}
            />
            <div className="ratingMessage">{ratingMessage}</div>
          </div>
          <div id="recommend">
            <label>
              Do you recommend this product?*
              <div onChange={this.handleRecommend}>
                <label>
                  Yes
                  <input type="radio" value={true} name="recommend" required />
                </label>
                <label>
                  No
                  <input type="radio" value={false} name="recommend" />
                </label>
              </div>
            </label>
          </div>
          <div id="characteristics">
            <label>
              Characteristics*
              <div>
                <SizeComponent handleChange={this.handleSize} meta={Size} />
                <WidthComponent handleChange={this.handleWidth} meta={Width} />
                <ComfortComponent
                  handleChange={this.handleComfort}
                  meta={Comfort}
                />
                <QualityComponent
                  handleChange={this.handleQuality}
                  meta={Quality}
                />
                <LengthComponent
                  handleChange={this.handleLength}
                  meta={Length}
                />
                <FitComponent handleChange={this.handleFit} meta={Fit} />
              </div>
            </label>
          </div>
          <div id="summary">
            <label>
              <div>Review summary</div>
              <input
                maxLength="60"
                onChange={this.handleSummary}
                value={summary}
                placeholder="Example: Best purchase ever!"
                type="text"
              />
            </label>
          </div>
          <div id="reviewBody">
            <label>
              <div>Reviewbody*</div>
              <textarea
                maxLength="1000"
                onChange={this.handleBody}
                value={body}
                placeholder="Why did you like the product or not?"
                minLength="50"
                required
              />
              <div>
                {50 - body.length <= 0
                  ? 'Minimum reached'
                  : `Minimum required characters left: ${50 - body.length}`}
              </div>
            </label>
          </div>
          <div id="reviewPhoto">
            <label>
              <div>Include a photo</div>
              <input
                onChange={this.handlePhoto}
                value={photo}
                placeholder="Paste URL"
                type="text"
              />
            </label>
          </div>
          <div id="reviewName">
            <label>
              <div>Name to display*</div>
              <input
                onChange={this.handleName}
                value={name}
                maxLength="60"
                placeholder="Example: jackson11!"
                type="text"
                required
              />
            </label>
          </div>
          <div id="reviewEmail">
            <label>
              <div>Your email*</div>
              <input
                onChange={this.handleEmail}
                value={email}
                maxLength="60"
                placeholder="Example: jacson11@email.com"
                type="text"
                required
              />
              <div id="authenticationMessage">
                For authentication reasons, you will not be emailed
              </div>
            </label>
          </div>
          <button type="button" onClick={toggle} id="cancelButton">
            CANCEL
          </button>
          <input type="submit" value ="SUBMIT" onClick={this.handleSubmit} id="submitButton" />

        </form>
      </div>
    );
  }
}

export default AddAReviewForm;
