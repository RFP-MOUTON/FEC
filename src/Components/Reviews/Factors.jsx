import React from 'react';

const Factors = () => {
  return (
    <div id="factors" >Factors</div>
  );
};

export default Factors;
