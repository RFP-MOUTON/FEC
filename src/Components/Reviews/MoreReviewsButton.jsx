import React from 'react';

const MoreReviewsBtn = ({ totalReviews, currentlyDisplayed, handleClick }) => {
  if (
    totalReviews.length <= 2 ||
    totalReviews.length - currentlyDisplayed.length === 0
  ) {
    return null;
  }
  return <button id="moreReviewsButton" type="button" onClick={handleClick}>MORE REVIEWS</button>;
};

export default MoreReviewsBtn;
