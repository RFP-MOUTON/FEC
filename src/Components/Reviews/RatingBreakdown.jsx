import React from 'react';

const RatingBreakdown = () => {
  return (
    <div>
      <div id="ratingBreakdown">Rating Breakdown</div>
    </div>
  );
};

export default RatingBreakdown;
