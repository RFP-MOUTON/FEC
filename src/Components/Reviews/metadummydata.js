const meta = {
  product_id: '17762',
  ratings: {
    1: '1',
    2: '1',
    3: '1',
    4: '1',
    5: '1',
  },
  recommended: {
    false: '0',
    true: '0',
  },
  characteristics: {},
};

export default meta;
