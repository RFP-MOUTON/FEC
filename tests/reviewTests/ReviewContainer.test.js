import React from 'react';
import { shallow } from 'enzyme';

import Reviews from '../../src/Components/ReviewsContainer.jsx';

describe('Review Container Componenet', () => {
  it('should render correctly', () => {
    const component = shallow(<Reviews />);

    expect(component).toMatchSnapshot();
  });
});
