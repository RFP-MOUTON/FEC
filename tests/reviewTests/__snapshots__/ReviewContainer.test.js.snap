// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Container Componenet should render correctly 1`] = `ShallowWrapper {}`;
